internal/hub/model/node_activity.go

```diff
@@ -224,3 +226,98 @@ func BuildPath(path string, query any, nodes []Cache) (map[common.Address]string
 
 	return urls, nil
 }
+
+type DataResponse struct {
+	Address common.Address
+	Data    []byte
+	// Valid indicates whether the data is non-null.
+	Valid          bool
+	Err            error
+	Request        int
+	InvalidRequest int
+}
+
+type ErrResponse struct {
```

Comment:
should provide a generic `ErrorResponse` struct

Reply:
this depends on the error responses returned by the node 
https://github.com/RSS3-Network/Node/blob/develop/common/http/response/error.go#L20